### ISSUE TITLE
Media synchronisation resets changes on metadata

### ISSUE THREAD

**Issue by Abqehebter** [AzuraCast/AzuraCast#3553]
**Using Docker installation method**
Yes

**AzuraCast version**
Rolling Release #6671882 (2020-12-11 22:53)

**Host Operating System**
Debian 9

**Describe the bug**
As soon as I upload a normal song, everything works as usual. If I change the metadata (artist, title, etc.) and save, the 5-minute sync changes the metadata back to the previous data.
The solution would be to enter the data manually and then upload the song, but I don't think that's the point.

**To Reproduce**
1. Go to a station
2. Music Files
3. Upload a Song
4. Edit the metadata to something else...
5. Go to the administration panel
6. Click on 'Run Task' by '5-minute sync'
7. See was happend

**Relevant Logs**
```
Loading...[2020-12-12T20:34:39.212231+00:00] AzuraCast.NOTICE: Running sync task: 5-Minute Sync [] []
[2020-12-12T20:34:39.212602+00:00] AzuraCast.DEBUG: Successfully acquired the "{resource}" lock. {"resource":{"Symfony\\Component\\Lock\\Key":"lock_sync_medium"}} []
[2020-12-12T20:34:39.212880+00:00] AzuraCast.DEBUG: Expiration defined for "{resource}" lock for "{ttl}" seconds. {"resource":{"Symfony\\Component\\Lock\\Key":"lock_sync_medium"},"ttl":600.0} []
[2020-12-12T20:34:39.311323+00:00] AzuraCast.INFO: Processing media for storage location Local: /var/azuracast/stations/test/media... [] []
[2020-12-12T20:34:39.319088+00:00] AzuraCast.INFO: Sending message {class} with {alias} sender using {sender} {"message":{"App\\Message\\ReprocessMediaMessage":{"media_id":1047,"force":false}},"class":"App\\Message\\ReprocessMediaMessage","alias":0,"sender":"Symfony\\Component\\Messenger\\Bridge\\Doctrine\\Transport\\DoctrineTransport"} []
[2020-12-12T20:34:39.319647+00:00] AzuraCast.DEBUG: Media processed for "Local: /var/azuracast/stations/test/media". {"total_size":"485192993 (485.2 MB)","total_files":70,"already_queued":0,"unchanged":69,"updated":1,"created":0,"deleted":0,"not_processable":0} []
[2020-12-12T20:34:39.371625+00:00] AzuraCast.DEBUG: Timer "CheckMediaTask" completed in 0.156 second(s). [] []
[2020-12-12T20:34:39.373270+00:00] AzuraCast.INFO: Processing auto-assigning folders for station... {"station":"Test"} []
[2020-12-12T20:34:39.422005+00:00] AzuraCast.INFO: Processing auto-assigning folders for station... {"station":"Test2"} []
[2020-12-12T20:34:39.475428+00:00] AzuraCast.INFO: Processing auto-assigning folders for station... {"station":"Test3"} []
[2020-12-12T20:34:39.550076+00:00] AzuraCast.INFO: Processing auto-assigning folders for station... {"station":"Test4"} []
[2020-12-12T20:34:39.619761+00:00] AzuraCast.INFO: Processing auto-assigning folders for station... {"station":"Test5"} []
[2020-12-12T20:34:39.905091+00:00] AzuraCast.INFO: Processing auto-assigning folders for station... {"station":"Test6"} []
[2020-12-12T20:34:39.935188+00:00] AzuraCast.DEBUG: Timer "CheckFolderPlaylistsTask" completed in 0.563 second(s). [] []
[2020-12-12T20:34:40.176422+00:00] AzuraCast.DEBUG: HTTP client POST call to https://central.azuracast.com/api/update produced response 200 [] []
[2020-12-12T20:34:40.176579+00:00] AzuraCast.INFO: Successfully checked for updates. {"results":{"needs_rolling_update":true,"rolling_updates_available":1,"rolling_updates_list":["Ansible warning in Bug Report (#3548)"],"needs_release_update":false,"current_release":"0.9.6.5","latest_release":"0.9.6.5"}} []
[2020-12-12T20:34:40.235619+00:00] AzuraCast.DEBUG: Timer "CheckUpdatesTask" completed in 0.300 second(s). [] []
```

**Comment by Takumi86ES** [comment]
This happened to me today.

Apparently it is a mistake.

**Comment by Abqehebter** [comment]
I assume so. Strangely enough, it's only in my folders. The change remains in the root directory.

**Comment by SlvrEagle23** [comment]
@Abqehebter Are you using a remote storage location (like S3 or Dropbox), or the local filesystem?

**Comment by Abqehebter** [comment]
I use the local filesystem

**Comment by kieranlittle667** [comment]
I have experienced this also. Docker installation on ubuntu 20.04, files stored locally.

**Comment by Takumi86ES** [comment]
local filesystem

**Comment by SlvrEagle23** [comment]
@Abqehebter I have found the source of this problem; the `getid3` library would return `false` when updating metadata even if the update was largely successful if it had _any_ portions of the metadata that it couldn't write; this is actually more of a "warning" on our end than a critical error or a failure to write to the file.

I've updated our handling of this error to properly warn of this error in the logs (for debug purposes) but to continue writing the changes, and more importantly to update the file modified time, which will prevent reprocessing of the file.

This should now be available on the rolling release channel.
